### PR TITLE
"Share all future addresses" storage

### PR DESCRIPTION
**Pull request by vitvly**
Related to https://github.com/status-im/status-mobile/issues/18976

This is a rebased version of work done by @cammellos in https://github.com/status-im/status-go/tree/feature/edit-permissions

### CODE REVIEW

**Comment by vitvly** [comment]
Tests issue related to https://github.com/status-im/status-go/commit/36273bc9b2b0a148d4e2439811a211e70391c14e

**Comment by vitvly** [comment]
`TestMessengerCommunitiesTokenPermissionsSuite/TestReevaluateMemberTokenMasterRoleInClosedCommunity_ERC20` this test (from run https://ci.status.im/job/status-go/job/prs/job/tests/job/PR-5354/7/display/redirect) might be flaky, as it passes locally (?)

**Comment by vitvly** [comment]
This has been also verified in a mobile PR: https://github.com/status-im/status-mobile/pull/20549#issuecomment-2231095293

**Review by cammellos — APPROVED** [review]

File: appdatabase/migrations/bindata.go [hunk 1]
```diff
@@ -118,6 +118,8 @@
 // 1720557751_rename_staging.up.sql (3.258kB)
 // 1720606449_enable_store_confirmation_for_messages_sent.up.sql (191B)
 // doc.go (94B)
+// exit_code_1.txt (2B)
+// test_1.log (56B)
```
Comment (osmaczko) [inline appdatabase/migrations/bindata.go]: Oops

File: static/bindata.go [hunk 2]
```diff
@@ -103,7 +103,7 @@ func emojisTxt() (*asset, error) {
 		return nil, err
 	}
 
-	info := bindataFileInfo{name: "emojis.txt", size: 28134, mode: os.FileMode(0664), modTime: time.Unix(1700000000, 0)}
+	info := bindataFileInfo{name: "emojis.txt", size: 28134, mode: os.FileMode(0644), modTime: time.Unix(1700000000, 0)}
```
Comment (osmaczko) [inline static/bindata.go]: :thinking:

**Review by osmaczko — APPROVED** [review]

File: appdatabase/migrations/bindata.go (hunk 1 above)
Comment (vitvly) [reply appdatabase/migrations/bindata.go]: Yes, these are some go test files that for some reason were picked up by `make generate`.

Comment (vitvly) [reply appdatabase/migrations/bindata.go]: Thanks for pointing this out, removed.

File: static/bindata.go (hunk 2 above)
Comment (vitvly) [reply static/bindata.go]: This i have no idea about, hmm.

Comment (vitvly) [reply static/bindata.go]: Manually reverted to older `FileMode` values.

**Pull request merged**